### PR TITLE
[Dashboard] Remove <Code /> component (ez mode)

### PR DESCRIPTION
**Pull request by kien-ngo**
## Problem solved

Short description of the bug fixed or feature added

---

## PR-Codex overview
This PR focuses on enhancing the user interface by replacing instances of `<Code>` components with a new `<InlineCode>` component for better styling and consistency across the application.

### Detailed summary
- Added `InlineCode` component to `inline-code.tsx`.
- Replaced `<Code>` with `<InlineCode>` in various components for:
  - `ChainValidation`
  - `EngineCorsConfig`
  - `EngineIpAllowlistConfig`
  - `InteractiveAbiFunction`
  - `EngineAccessTokens`
  - `SnapshotUpload`
  - `UploadStep`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by kien-ngo** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/4787?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#4787** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/4787?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#4786** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/4786?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4785** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/4785?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4784** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/4784?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4765** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/4765?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4750** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/4750?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4749** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/4749?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4748** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/4748?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @kien-ngo and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by MananTank — CHANGES_REQUESTED** [review]
I don't think unstyled code element looks similar to Chakra Code component. Can you verify? Most likely we'll need to add background, a bit of padding, rounded corners etc

**Comment by MananTank** [comment]
We can make a `InlineCode` component and use it everywhere

**Comment by jnsdls** [comment]
> We can make a `InlineCode` component and use it everywhere 

we should do exactly this

File: apps/dashboard/src/core-ui/batch-upload/upload-step.tsx [hunk 1]
```diff
@@ -79,6 +88,7 @@ export const UploadStep: React.FC<UploadStepProps> = ({
                   Download example.json
                 </Link>
                 .
+<<<<<<< HEAD
```
Comment (jnsdls) [inline apps/dashboard/src/core-ui/batch-upload/upload-step.tsx:91]: merge conflicts

**Review by jnsdls — CHANGES_REQUESTED** [review]

File: apps/dashboard/src/@/components/ui/inline-code.tsx [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import { cn } from "@/lib/utils";
+
+export function InlineCode({
+  code,
+  className,
+}: { code: string; className?: string }) {
+  return (
+    <code
+      className={cn(
+        "inline-block rounded bg-muted px-2 font-mono text-xs",
```
Comment (MananTank) [inline apps/dashboard/src/@/components/ui/inline-code.tsx:10]: text-xs looks a bit too small - lets change it to text-sm

File: apps/dashboard/src/contract-ui/tabs/permissions/page.tsx [hunk 3]
```diff
@@ -1,7 +1,9 @@
 "use client";
 
+import { InlineCode } from "@/components/ui/inline-code";
 import { useIsomorphicLayoutEffect } from "@/lib/useIsomorphicLayoutEffect";
-import { ButtonGroup, Code, Divider, Flex } from "@chakra-ui/react";
+import {} from "@chakra-ui/react";
```
Comment (jnsdls) [inline apps/dashboard/src/contract-ui/tabs/permissions/page.tsx:5]: remove empty import

**Review by jnsdls — CHANGES_REQUESTED** [review]

File: apps/dashboard/src/contract-ui/tabs/permissions/page.tsx (hunk 3 above)
Comment (kien-ngo) [reply apps/dashboard/src/contract-ui/tabs/permissions/page.tsx:5]: oops. I trusted the linter to do its job!!

File: apps/dashboard/src/contract-ui/tabs/claim-conditions/components/snapshot-upload.tsx [hunk 4]
```diff
@@ -266,8 +266,11 @@ export const SnapshotUpload: React.FC<SnapshotUploadProps> = ({
                       <>
                         <Text as={ListItem}>
                           Files <em>must</em> contain one .csv file with a list
-                          of addresses and their <Code>maxClaimable</Code>.
-                          (amount each wallet is allowed to claim)
+                          of addresses and their{" "}
+                          <code className="rounded bg-gray-200/20 p-1 text-black dark:text-white">
```
Comment (MananTank) [inline apps/dashboard/src/contract-ui/tabs/claim-conditions/components/snapshot-upload.tsx:270]: Also use the InlineCode component here

Comment (kien-ngo) [reply apps/dashboard/src/contract-ui/tabs/claim-conditions/components/snapshot-upload.tsx:270]: urgg nice catch ser. it was my demo code

**Review by MananTank — APPROVED** [review]

**Review by jnsdls — APPROVED** [review]

**Pull request merged**